cmd/src/batch_common.go

```diff
@@ -604,3 +583,14 @@ func getBatchParallelism(ctx context.Context, flag int) (int, error) {
 
 	return docker.NCPU(ctx)
 }
+
+func validateSourcegraphVersionConstraint(ctx context.Context, svc *service.Service) error {
+	ffs, err := svc.DetermineFeatureFlags(ctx)
+	if err != nil {
+		return err
+	}
+	if ffs.Sourcegraph40 {
+		return nil
+	}
+	return errors.Newf("\n\n * Warning:\n This version of src-cli requires Sourcegraph version 4.0 or newer. If you're not on Sourcegraph 4.0 or newer, please use any 3.x release of src-cli.\n\n")
```

Comment:
While I doubt anyone is _really_ going to language lawyer this into complaining that src-cli 3.2.0 doesn't work with Sourcegraph 3.43, how about:

```suggestion
	return errors.Newf("\n\n * Warning:\n This version of src-cli requires Sourcegraph version 4.0 or newer. If you're not on Sourcegraph 4.0 or newer, please use the 3.x release of src-cli that corresponds to your Sourcegraph version.\n\n")
```